### PR TITLE
[Magiclysm] Adding more attunement spells

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
Mods "Adding more attunement spells"
#### Purpose of change
List of changes:
Alchemist: Manafusion bomb and Manafusion trap spells, 
items for it (need to make a normal recipe for another classes, but with higher price), 
replasing AURA_MAGE attunement to ALCHEMIST, 
added description to alchemist attunement
Boreal mage: Auroral Guard spell + ethereal item for it + material for it (it was used also in Crusader spells, but now there is another person that work on it)
Cleansing Flame - Hermes Reborn spell
Gaias Chosen - Samsara spell
Ice Elemental - Cold Heart spell + effect type for it
Magnetism Mage - Electrolaser spell
Sun Mage - Pocket Sun spell + ethereal item for it
Void Mage - Final Grasp and Horizon Vortex spells
Vulcanist - Pyroclastic Flow Spell
Wither Mage - Life Drain spell
#### Additional context
[More info can be here](https://github.com/CleverRaven/Cataclysm-DDA/pull/48045)

### CODE REVIEW

**Comment by GuardianDll** [comment]
Crap. Sorry, i dont want to waste my time for some dumb splitting of not so big 1k string json change, when each branch that i create want to contain each previous update, which make me angry, and my chair burnt. I'll better spend my time making some another interesting attunement stuff instead of this.

**Comment by actual-nh** [comment]
Would you like me to go over them? I'm not sure what you're intending.

**Comment by actual-nh** [comment]
I will check it over when I have a moment - possibly this weekend. @BrettDong: This unfortunately needs String Freeze (and Content Freeze).

File: data/mods/Magiclysm/Spells/attunements/Alchemist.json [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    "final_energy_cost": 1200,
+    "energy_increment": 20,
+    "extra_effects": [ { "id": "alchemist_make_potion_bomb_create" } ],
+    "difficulty": 8
+  },
+  {
+    "id": "alchemist_make_potion_bomb_create",
+    "type": "SPELL",
+    "name": "Manafusion bomb create",
+    "description": "Gives you a manafusion bomb. Subspell; you aren't allowed to see it except in debug mode.",
```
Comment (BrettDong) [inline data/mods/Magiclysm/Spells/attunements/Alchemist.json:33]:
```suggestion
    "description": "Gives you a manafusion bomb.  Subspell; you aren't allowed to see it except in debug mode.",
```

File: data/mods/Magiclysm/Spells/attunements/Alchemist.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    "final_energy_cost": 1200,
+    "energy_increment": 20,
+    "extra_effects": [ { "id": "alchemist_make_potion_trap_create" } ],
+    "difficulty": 8
+  },
+  {
+    "id": "alchemist_make_potion_trap_create",
+    "type": "SPELL",
+    "name": "Manafusion trap create",
+    "description": "Gives you a manafusion trap. Subspell; you aren't allowed to see it except in debug mode.",
```
Comment (BrettDong) [inline data/mods/Magiclysm/Spells/attunements/Alchemist.json:73]:
```suggestion
    "description": "Gives you a manafusion trap.  Subspell; you aren't allowed to see it except in debug mode.",
```

File: data/mods/Magiclysm/Spells/attunements/Cleansing_Flame.json [hunk 3]
```diff
@@ -75,5 +75,54 @@
...
+    "damage_increment": -2.5,
+    "max_level": 40,
+    "extra_effects": [ { "id": "cleansingflame_hermes_burn" }, { "id": "cleansingflame_hermes_split" } ],
+    "flags": [ "SOMATIC", "VERBAL", "CONCENTRATE", "LOUD", "MUST_HAVE_CLASS_TO_LEARN" ]
+  },
+  {
+    "id": "cleansingflame_hermes_burn",
+    "type": "SPELL",
+    "name": "Hermes Reborn fire",
+    "description": "Give some frost damage for Hermes Reborn spell. Subspell; you aren't allowed to see it except in debug mode.",
```
Comment (BrettDong) [inline data/mods/Magiclysm/Spells/attunements/Cleansing_Flame.json:103]:
```suggestion
    "description": "Give some frost damage for Hermes Reborn spell.  Subspell; you aren't allowed to see it except in debug mode.",
```

File: data/mods/Magiclysm/Spells/attunements/Cleansing_Flame.json [hunk 4]
```diff
@@ -75,5 +75,54 @@
...
+    "min_aoe": 3,
+    "max_aoe": 5,
+    "aoe_increment": 0.05,
+    "spell_class": "CLEANSING_FLAME"
+  },
+  {
+    "id": "cleansingflame_hermes_split",
+    "type": "SPELL",
+    "name": "Hermes Reborn Pain Split",
+    "description": "Same as Biomancer's Pain Split. Subspell; you aren't allowed to see it except in debug mode.",
```
Comment (BrettDong) [inline data/mods/Magiclysm/Spells/attunements/Cleansing_Flame.json:122]:
```suggestion
    "description": "Same as Biomancer's Pain Split.  Subspell; you aren't allowed to see it except in debug mode.",
```

File: data/mods/Magiclysm/Spells/attunements/Gaias_Chosen.json [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    "damage_increment": -4.5,
+    "max_level": 40,
+    "extra_effects": [ { "id": "gaiaschosen_samsara_cureall", "hit_self": true } ],
+    "flags": [ "SOMATIC", "VERBAL", "CONCENTRATE", "LOUD", "MUST_HAVE_CLASS_TO_LEARN" ]
+  },
+  {
+    "id": "gaiaschosen_samsara_cureall",
+    "type": "SPELL",
+    "name": "Samsara Cureall effect",
+    "description": "Gives you a cure effect. Subspell; you aren't allowed to see it except in debug mode.",
```
Comment (BrettDong) [inline data/mods/Magiclysm/Spells/attunements/Gaias_Chosen.json:29]:
```suggestion
    "description": "Gives you a cure effect.  Subspell; you aren't allowed to see it except in debug mode.",
```

File: data/mods/Magiclysm/Spells/attunements/Ice_Elemental.json [hunk 6]
```diff
@@ -22,5 +22,48 @@
...
+    "base_energy_cost": 80,
+    "energy_increment": -3,
+    "final_energy_cost": 200,
+    "extra_effects": [ { "id": "cold_heart_slow" } ]
+  },
+  {
+    "id": "cold_heart_slow",
+    "type": "SPELL",
+    "name": "Cold Heart slowdown",
+    "description": "Causes slowdown from the Cold Heart spell. Subspell; you aren't allowed to see it except in debug mode.",
```
Comment (BrettDong) [inline data/mods/Magiclysm/Spells/attunements/Ice_Elemental.json:58]:
```suggestion
    "description": "Causes slowdown from the Cold Heart spell.  Subspell; you aren't allowed to see it except in debug mode.",
```

File: data/mods/Magiclysm/Spells/attunements/Void_Mage.json [hunk 7]
```diff
@@ -0,0 +1,229 @@
+[
+  {
+    "type": "SPELL",
+    "id": "voidmage_final_grasp",
+    "name": "Final Grasp",
+    "description": "In the end, everything succumbs to the grasp of nonexistence and stasis. This spell gives the target a foretaste of that.",
```
Comment (BrettDong) [inline data/mods/Magiclysm/Spells/attunements/Void_Mage.json:6]:
```suggestion
    "description": "In the end, everything succumbs to the grasp of nonexistence and stasis.  This spell gives the target a foretaste of that.",
```

File: data/mods/Magiclysm/items/alchemy_items.json [hunk 8]
```diff
@@ -291,5 +291,125 @@
     "bashing": 10,
     "to_hit": 1,
     "category": "spare_parts"
+  },
+  {
+    "id": "alchemist_bomb",
+    "type": "TOOL",
+    "category": "weapons",
+    "name": { "str": "Мana bomb" },
+    "description": "Mana bombs are powerful explosive devices developed by a little-known alchemist group for use against enemy spellcasters. These bombs are fueled by pure arcane energy and can kill instantly. The exact workings are unclear, but they appear to use some kind of massive mana burn.",
```
Comment (BrettDong) [inline data/mods/Magiclysm/items/alchemy_items.json:300]:
```suggestion
    "description": "Mana bombs are powerful explosive devices developed by a little-known alchemist group for use against enemy spellcasters.  These bombs are fueled by pure arcane energy and can kill instantly.  The exact workings are unclear, but they appear to use some kind of massive mana burn.",
```

File: data/mods/Magiclysm/items/alchemy_items.json [hunk 9]
```diff
@@ -291,5 +291,125 @@
...
+      "type": "transform"
+    },
+    "flags": [ "BOMB", "GRENADE" ]
+  },
+  {
+    "id": "alchemist_bomb_act",
+    "type": "TOOL",
+    "category": "weapons",
+    "name": { "str": "mana bomb (lit)", "str_pl": "mana bombs (lit)" },
+    "description": "The liquid in the bottle is sizzling. Throwing it away from you, preferably at an enemy, would be a very good idea.",
```
Comment (BrettDong) [inline data/mods/Magiclysm/items/alchemy_items.json:326]:
```suggestion
    "description": "The liquid in the bottle is sizzling.  Throwing it away from you, preferably at an enemy, would be a very good idea.",
```

**Review by BrettDong — COMMENTED** [review]
The convention is to put two spaces between sentences.

File: data/mods/Magiclysm/Spells/attunements/Alchemist.json [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+    "effect_str": "alchemist_bomb",
+    "shape": "blast",
+    "min_damage": 1,
+    "max_damage": 1
+  },
+  {
+    "id": "alchemist_make_potion_trap",
+    "type": "SPELL",
+    "name": "Manafusion trap",
+    "description": "Made manafusion trap, stabilizing reaction and creating mana.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/Spells/attunements/Alchemist.json:46]:
```suggestion
    "description": "Make a manafusion trap, stabilizing the reaction by drawing off excess mana.",
```

File: data/mods/Magiclysm/Spells/attunements/Void_Mage.json [hunk 11]
```diff
@@ -0,0 +1,229 @@
...
+    "energy_source": "MANA",
+    "base_energy_cost": 300,
+    "difficulty": 9,
+    "extra_effects": [ { "id": "voidmage_final_grasp_slowdown" } ]
+  },
+  {
+    "id": "voidmage_final_grasp_slowdown",
+    "type": "SPELL",
+    "name": "Final Grasp slowdown",
+    "description": "Removes some moves from enemy. Subspell; you aren't allowed to see it except in debug mode.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/Spells/attunements/Void_Mage.json:35]:
```suggestion
    "description": "Removes some moves from an enemy. Subspell; you aren't allowed to see it except in debug mode.",
```

File: data/mods/Magiclysm/Spells/attunements/Boreal_Mage.json [hunk 12]
```diff
@@ -48,5 +48,25 @@
     "energy_source": "MANA",
     "damage_type": "cold",
     "max_level": 35
+  },
+  {
+    "id": "borealmage_auroral_guard",
+    "type": "SPELL",
+    "name": { "str": "Auroral Guard" },
+    "description": "Create sturdy shell what wrap your body and gain almost perfect defence.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/Spells/attunements/Boreal_Mage.json:56]:
```suggestion
    "description": "Create a sturdy shell that wraps your body, gaining a much better defense.",
```
It isn't a perfect defense; that would be invulnerable and not have any encumbrance...

**Review by actual-nh — COMMENTED** [review]
A few edits; need to do other things now, but didn't want to lose these.

File: data/mods/Magiclysm/Spells/attunements/Sun_Mage.json [hunk 13]
```diff
@@ -29,5 +29,30 @@
     "max_level": 35,
     "base_casting_time": 350,
     "energy_source": "MANA"
+  },
+  {
+    "id": "sunmage_pocket_sun",
+    "type": "SPELL",
+    "name": "Pocket Sun",
+    "description": "You create a stable gap on the fabric of reality near the sun, and can use his light and temperature for own purposes.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/Spells/attunements/Sun_Mage.json:37]:
```suggestion
    "description": "You create a stable gap in the fabric of reality near the sun, through which you can draw on its light and temperature.",
```

File: data/mods/Magiclysm/Spells/attunements/Ice_Elemental.json [hunk 14]
```diff
@@ -22,5 +22,48 @@
     "min_duration": 18000,
     "max_duration": 120000,
     "duration_increment": 4500
+  },
+  {
+    "id": "cold_heart",
+    "type": "SPELL",
+    "name": "Cold Heart",
+    "description": "Harshly drop the temperature of your enemies from the inside, slowing down his bloodstream or even shattering organs.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/Spells/attunements/Ice_Elemental.json:30]: Again, I seem to recall having rewritten this one before?

File: data/mods/Magiclysm/Spells/attunements/Void_Mage.json [hunk 15]
```diff
@@ -0,0 +1,229 @@
...
+    "max_level": 40,
+    "min_damage": -100,
+    "max_damage": -1000,
+    "damage_increment": -23
+  },
+  {
+    "type": "SPELL",
+    "id": "voidmage_horizon_vortex",
+    "name": "Horizon Vortex",
+    "description": "Creates a chaotic energy field, that shutter enemies inside it. Its pretty random, and can either kill everyone in area or produce almost nothing.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/Spells/attunements/Void_Mage.json:48]: Again, I seem to recall having rewritten this one before?

File: data/mods/Magiclysm/Spells/attunements/Wither_Mage.json [hunk 16]
```diff
@@ -52,5 +52,46 @@
...
+    "final_energy_cost": 250,
+    "energy_increment": -8.6,
+    "difficulty": 10,
+    "extra_effects": [ { "id": "life_drain_stamina", "hit_self": true } ]
+  },
+  {
+    "type": "SPELL",
+    "id": "life_drain_stamina",
+    "name": "Life Drain stamina",
+    "description": "Gives you a bit of stamina. Subspell; you aren't allowed to see it except in debug mode.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/Spells/attunements/Wither_Mage.json:86]:
```suggestion
    "description": "Gives you a bit of stamina.  Subspell; you aren't allowed to see it except in debug mode.",
```

File: data/mods/Magiclysm/items/alchemy_items.json [hunk 17]
```diff
@@ -291,5 +291,125 @@
...
+      "no_deactivate_msg": "The reaction has already started - all you can do is get it away from you!",
+      "explosion": { "power": 200 }
+    },
+    "flags": [ "TRADER_AVOID" ]
+  },
+  {
+    "id": "alchemist_mine",
+    "type": "TOOL",
+    "name": { "str": "alchemist_mine" },
+    "description": "Pair of alchemist bombs with concentrated mana wire, reaction is stable, but any movement in mana wire zone provoke extremely powerful explosion in small area.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/items/alchemy_items.json:355]: Again, I seem to recall having rewritten this one before?

File: data/mods/Magiclysm/items/ethereal_items.json [hunk 18]
```diff
@@ -771,5 +771,63 @@
     "relic_data": {
       "passive_effects": [ { "has": "WIELD", "condition": "ALWAYS", "values": [ { "value": "ITEM_DAMAGE_PURE", "add": 10 } ] } ]
     }
+  },
+  {
+    "id": "aura_auroral_guard",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": { "str_sp": "auroral shell" },
+    "description": "Sturdy shell what wrap your body and gain almost perfect defence.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/items/ethereal_items.json:780]:
```suggestion
    "description": "A sturdy shell that wraps your body, giving you a much better defense.",
```

File: data/mods/Magiclysm/items/ethereal_items.json [hunk 19]
```diff
@@ -771,5 +771,63 @@
...
+    "description": "Small capsule-like field of solar energy.",
+    "weight": "1 g",
+    "volume": "1 ml",
+    "material": [ "sunlight" ],
+    "symbol": "*",
+    "color": "yellow",
+    "qualities": [ [ "BOIL", 2 ] ],
+    "use_action": {
+      "target": "pocket_sun_cover",
+      "msg": "You cover the stabilizing field, and now gap doesn't create light.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/items/ethereal_items.json:807]:
```suggestion
      "msg": "You cover the pocket sun, and now the gap doesn't emit light.",
```

File: data/mods/Magiclysm/items/ethereal_items.json [hunk 20]
```diff
@@ -771,5 +771,63 @@
...
+    "material_thickness": 1.7,
+    "environmental_protection": 8,
+    "flags": [ "ONLY_ONE", "OVERSIZE", "PERSONAL", "STURDY" ]
+  },
+  {
+    "id": "pocket_sun",
+    "type": "TOOL",
+    "category": "tools",
+    "name": { "str": "Pocket Sun" },
+    "description": "Small capsule-like field of solar energy.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/items/ethereal_items.json:798]:
```suggestion
    "description": "A small, capsule-shaped field containing a gap in reality that lets through solar energy.",
```

File: data/mods/Magiclysm/items/ethereal_items.json [hunk 21]
```diff
@@ -771,5 +771,63 @@
...
+      "type": "transform"
+    },
+    "flags": [ "LIGHT_700", "TRADER_AVOID" ]
+  },
+  {
+    "id": "pocket_sun_cover",
+    "type": "TOOL",
+    "category": "weapons",
+    "name": { "str": "Covered Pocket Sun" },
+    "description": "Temporary covered capsule-like field of solar energy.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/items/ethereal_items.json:818]:
```suggestion
    "description": "A small, capsule-shaped field containing a gap in reality that lets through solar energy; it is currently covered.",
```

File: data/mods/Magiclysm/items/ethereal_items.json [hunk 22]
```diff
@@ -771,5 +771,63 @@
...
+    "weight": "1 g",
+    "volume": "1 ml",
+    "material": [ "sunlight" ],
+    "symbol": "*",
+    "color": "yellow",
+    "qualities": [ [ "BOIL", 2 ] ],
+    "use_action": {
+      "target": "pocket_sun",
+      "msg": "You relieve the cover on stabilizing field, and gap start shining now.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/items/ethereal_items.json:827]:
```suggestion
      "msg": "You remove the cover on the pocket sun, and now the gap emits light again.",
```

File: data/mods/Magiclysm/traits/attunements.json [hunk 23]
```diff
@@ -41,15 +41,16 @@
...
+    "id": "ALCHEMIST",
     "type": "mutation",
-    "name": { "str": "Aura Mage" },
+    "name": { "str": "Alchemist" },
     "points": 8,
     "starting_trait": false,
     "purifiable": false,
     "valid": false,
-    "description": "",
+    "description": "Mix of worlds life energy from animist and technomancer's slant of work create an alchemist.  It requires magical talent and lots of time, but is invaluable for creating powerful magical effects in a storable, accessible form.",
```
Comment (actual-nh) [inline data/mods/Magiclysm/traits/attunements.json:51]: I'm not sure about "world's life energy" with regard to technomancers... will copyedit once clarified.

**Review by actual-nh — COMMENTED** [review]
BTW, I can see at least a few parts of this PR that can be made independent (e.g., changing "AURA_MAGE" to "ALCHEMIST") - will try to take a look at this week or (more likely) next weekend.

File: data/mods/Magiclysm/Spells/attunements/Void_Mage.json (hunk 15 above)
Comment (casswedson) [reply data/mods/Magiclysm/Spells/attunements/Void_Mage.json:48]:
```diff
-    "description": "Creates a chaotic energy field, that shutter enemies inside it.  Its pretty random, and can either kill everyone in area or produce almost nothing.",
+    "description": "Creates a chaotic energy field that disrupts enemies inside it.  It is rather random, and its effects can range from almost nothing to killing all hostiles in the area.",
```
You mean this?

**Pull request closed without merging**